packages/executor/src/execution/mapAsyncIterable.ts

```diff
@@ -0,0 +1,53 @@
+import { PromiseOrValue } from './types';
+
+/**
+ * Given an AsyncIterable and a callback function, return an AsyncIterator
+ * which produces values mapped via calling the callback function.
+ */
+export function mapAsyncIterable<T, U, R = undefined>(
```

Comment:
Maybe `mapAsyncIterator` can do the same work by taking the iterator like the following line.